### PR TITLE
feat: support to forcibly delete an AS group

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

* support to forcibly delete an AS group
* make scaling_configuration_id be required
* delete useless debug message

If the AS group has no configurations, it will be failed when activating it.
```
POST https://as.cn-north-4.myhuaweicloud.com/autoscaling-api/v1/0970dd7a1300f5672ff2c003c60ae115/scaling_group/645a3a5b-237f-4410-9473-065f2e1daf4e/action], error message: {"error":{"code":"AS.2018","message":"No AS configuration is in the AS group."}
```
It's better to make `scaling_configuration_id` be required. To keep compatibility， we just marked it in docs and schema json.

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccASGroup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccASGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccASGroup_basic
=== PAUSE TestAccASGroup_basic
=== CONT  TestAccASGroup_basic
--- PASS: TestAccASGroup_basic (250.24s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        250.316s

$ make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccASGroup_forceDelete'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccASGroup_forceDelete -timeout 360m -parallel 4
=== RUN   TestAccASGroup_forceDelete
=== PAUSE TestAccASGroup_forceDelete
=== CONT  TestAccASGroup_forceDelete
--- PASS: TestAccASGroup_forceDelete (199.25s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        199.315s

$ make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccASConfiguration_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccASConfiguration_basic -timeout 360m -parallel 4
=== RUN   TestAccASConfiguration_basic
=== PAUSE TestAccASConfiguration_basic
=== CONT  TestAccASConfiguration_basic
--- PASS: TestAccASConfiguration_basic (52.89s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        52.944s
```

### CODE REVIEW

File: huaweicloud/services/as/resource_huaweicloud_as_group.go [hunk 1]
```diff
@@ -397,32 +398,60 @@ func refreshInstancesLifeStates(asClient *golangsdk.ServiceClient, groupID strin
...
+		PollInterval: 10 * time.Second,
 	}
 
 	_, err := stateConf.WaitForStateContext(ctx)
 
 	return err
 }
 
-func checkASGroupInstancesRemoved(ctx context.Context, asClient *golangsdk.ServiceClient, groupID string, timeout time.Duration) error {
+func checkASGroupInstancesRemoved(ctx context.Context, client *golangsdk.ServiceClient, groupID string, timeout time.Duration) error {
```
Comment (niuzhenguo) [inline huaweicloud/services/as/resource_huaweicloud_as_group.go:433]: can we merge checkASGroupInstanceInService and checkASGroupInstancesRemoved into one function? Most of them are same.

File: huaweicloud/services/as/resource_huaweicloud_as_group.go [hunk 2]
```diff
@@ -397,32 +398,60 @@ func refreshInstancesLifeStates(asClient *golangsdk.ServiceClient, groupID strin
...
+
+func checkASGroupInstancesInService(ctx context.Context, client *golangsdk.ServiceClient, groupID string, insNum int, timeout time.Duration) error {
 	stateConf := &resource.StateChangeConf{
-		Pending: []string{"PENDING"},
-		Target:  []string{"INSERVICE"}, //if there is no lifecyclestatus, meaning no instances in asg
-		Refresh: refreshInstancesLifeStates(asClient, groupID, insNum, true),
-		Timeout: timeout,
-		Delay:   10 * time.Second,
+		Pending:      []string{"PENDING"},
+		Target:       []string{"INSERVICE"}, //if there is no lifecyclestatus, meaning no instances in asg
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/as/resource_huaweicloud_as_group.go]: this  comment looks useless and not in right place

File: huaweicloud/services/as/resource_huaweicloud_as_group.go [hunk 3]
```diff
@@ -397,32 +398,60 @@ func refreshInstancesLifeStates(asClient *golangsdk.ServiceClient, groupID strin
...
-func checkASGroupInstancesRemoved(ctx context.Context, asClient *golangsdk.ServiceClient, groupID string, timeout time.Duration) error {
+func checkASGroupInstancesRemoved(ctx context.Context, client *golangsdk.ServiceClient, groupID string, timeout time.Duration) error {
 	stateConf := &resource.StateChangeConf{
-		Pending: []string{"REMOVING"},
-		Target:  []string{""}, //if there is no lifecyclestatus, meaning no instances in asg
-		Refresh: refreshInstancesLifeStates(asClient, groupID, 0, false),
-		Timeout: timeout,
-		Delay:   10 * time.Second,
+		Pending:      []string{"REMOVING"},
+		Target:       []string{""}, //if there is no lifecyclestatus, meaning no instances in asg
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/as/resource_huaweicloud_as_group.go]: this comment not in right place

File: huaweicloud/services/as/resource_huaweicloud_as_group.go [hunk 4]
```diff
@@ -445,19 +474,11 @@ func resourceASGroupCreate(ctx context.Context, d *schema.ResourceData, meta int
...
+	logp.Printf("[DEBUG] instance number options: min(%d), max(%d), desired(%d)", minNum, maxNum, desireNum)
 	if desireNum < minNum || desireNum > maxNum {
-		return diag.Errorf("Invalid parameters: it should be min_instance_number<=desire_instance_number<=max_instance_number")
-	}
-	var initNum int
-	if desireNum > 0 {
-		initNum = desireNum
-	} else {
-		initNum = minNum
+		return diag.Errorf("invalid parameters: it should be min_instance_number<=desire_instance_number<=max_instance_number")
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/as/resource_huaweicloud_as_group.go]: add spaces around "<="

File: huaweicloud/services/as/resource_huaweicloud_as_group.go [hunk 5]
```diff
@@ -640,11 +654,9 @@ func resourceASGroupUpdate(ctx context.Context, d *schema.ResourceData, meta int
 		desireNum = minNum
 	}
 	if d.HasChanges("min_instance_number", "max_instance_number", "desire_instance_number") {
-		logp.Printf("[DEBUG] Min instance number is: %#v", minNum)
-		logp.Printf("[DEBUG] Max instance number is: %#v", maxNum)
-		logp.Printf("[DEBUG] Desire instance number is: %#v", desireNum)
+		logp.Printf("[DEBUG] instance number options: min(%d), max(%d), desired(%d)", minNum, maxNum, desireNum)
 		if desireNum < minNum || desireNum > maxNum {
-			return diag.Errorf("Invalid parameters: it should be min_instance_number<=desire_instance_number<=max_instance_number")
+			return diag.Errorf("invalid parameters: it should be min_instance_number<=desire_instance_number<=max_instance_number")
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/as/resource_huaweicloud_as_group.go]: add spaces around "<="

Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_group.go]: fixed

File: huaweicloud/services/as/resource_huaweicloud_as_group.go (hunk 4 above)
Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_group.go]: fixed

File: huaweicloud/services/as/resource_huaweicloud_as_group.go (hunk 2 above)
Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_group.go]: this comment will be deleted

File: huaweicloud/services/as/resource_huaweicloud_as_group.go (hunk 1 above)
Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_group.go:433]: the **resource.StateChangeConf** has many parameters (over 6), if we merge them into one function, we should define a function including those parameters. I don't think this is a good way to refactor it.

Also it's a common problem in our code, let's find a new way to refactor later.

Comment (niuzhenguo) [reply huaweicloud/services/as/resource_huaweicloud_as_group.go:433]: ok

**Pull request merged**